test/prism/parse_test.rb

```diff
@@ -69,11 +69,108 @@ def test_parse_lex
       assert_equal 5, tokens.length
     end
 
+    def test_dump_file
+      assert_nothing_raised do
+        Prism.dump_file(__FILE__)
+      end
+
+      error = assert_raise Errno::ENOENT do
+        Prism.dump_file("idontexist.rb")
+      end
+
+      assert_equal "No such file or directory - idontexist.rb", error.message
+
+      error = assert_raise Errno::EFAULT do
+        Prism.dump_file(nil)
+      end
+
+      assert_equal "Bad address", error.message
```

Comment:
It feels a bit strange that nil would raise EFAULT.
I would expect a coercion error, just like `File.read` does:
```
$ ruby -e 'File.read nil'
-e:1:in `read': no implicit conversion of nil into String (TypeError)
```

Reply:
It's now raising a `TypeError`.